weld/src/main/java/org/jboss/as/weld/deployment/processors/WeldDeploymentProcessor.java

```diff
@@ -299,6 +306,15 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
 
     }
 
+    private List<ServiceName> getJNDISubsytemDependencies() {
+        List<ServiceName> dependencies = new ArrayList<>();
+        dependencies.add(ContextNames.JBOSS_CONTEXT_SERVICE_NAME.append(ServiceName.of("UserTransaction")));
```

Comment:
This feels like a hack, there is not reason why these bindings should be treated differently to any other subsystem created bindings.

With that said though I guess this is ok for now, and it can be addressed properly later.